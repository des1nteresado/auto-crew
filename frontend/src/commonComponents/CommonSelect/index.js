import React from 'react';
import PropTypes from 'prop-types';
import { v1 as uuid } from 'uuid';
import { MenuItem } from '@material-ui/core';
import ExpandMoreRoundedIcon from '@material-ui/icons/ExpandMoreRounded';

import CommonInput from 'commonComponents/CommonInput';

const CommonSelect = ({ value, onChange, items, ...otherProps }) => {
  return (
    <CommonInput
      value={value}
      select
      SelectProps={{ IconComponent: ExpandMoreRoundedIcon }}
      onChange={onChange}
      {...otherProps}
    >
      {items.map((item) => (
        <MenuItem key={uuid()} value={item.value || item}>
          {item.label || item}
        </MenuItem>
      ))}
    </CommonInput>
  );
};

CommonSelect.propTypes = {
  items: PropTypes.oneOfType([PropTypes.array]).isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};

export default React.memo(CommonSelect);
