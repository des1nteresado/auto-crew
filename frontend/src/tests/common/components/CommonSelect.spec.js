import React from 'react';
import { shallow } from 'enzyme';

import CommonSelect from 'common/components/CommonSelect';

const defaultProps = {
  value: 'value',
  onChange: () => {},
  items: ['ITEM1', 'ITEM2'],
};

jest.mock('uuid', () => ({
  v1: () => 'test-id',
}));

const setUp = (props) => shallow(<CommonSelect {...defaultProps} {...props} />);

describe('CommonSelect', () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it('renders CommonSelect component correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders items prop correctly', () => {
    const menuItems = component.find('WithStyles(ForwardRef(MenuItem))');

    expect(menuItems).toHaveLength(defaultProps.items.length);

    menuItems.forEach((menuItem) => {
      expect(defaultProps.items.some((item) => item === menuItem.text())).toBeTruthy();
    });
  });
});
