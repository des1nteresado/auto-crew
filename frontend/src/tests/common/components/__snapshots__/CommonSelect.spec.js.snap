// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonSelect renders CommonSelect component correctly 1`] = `
<Memo(WithStyles(CommonInput))
  SelectProps={
    Object {
      "IconComponent": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "displayName": "ExpandMoreRoundedIcon",
        "muiName": "SvgIcon",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
    }
  }
  onChange={[Function]}
  select={true}
  value="value"
>
  <WithStyles(ForwardRef(MenuItem))
    key="test-id"
    value="ITEM1"
  >
    ITEM1
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="test-id"
    value="ITEM2"
  >
    ITEM2
  </WithStyles(ForwardRef(MenuItem))>
</Memo(WithStyles(CommonInput))>
`;
